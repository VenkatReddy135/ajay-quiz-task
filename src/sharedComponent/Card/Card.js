/* eslint-disable react/destructuring-assignment */
import React from "react";
import PropTypes from "prop-types";

export default function Card(props) {
  return (
    <div style={{ marginTop: "10em" }}>
      <div className={`card`} style={{ width: "20rem", margin: "auto" }}>
        <div className="card-body">
          <h2 className="card-title">{props.cardTitle}</h2>
          <div className="card-text">
            <>{props.cardText}</>
          </div>
        </div>
      </div>
    </div>
  );
}
Card.propTypes = {
  cardTitle: PropTypes.string,
  cardText: PropTypes.any,
};

Card.defaultProps = {
  cardTitle: "",
};
