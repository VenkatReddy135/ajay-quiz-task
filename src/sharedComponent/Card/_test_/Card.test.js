/* eslint-disable testing-library/no-node-access */
import React from "react";
import { cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Card from "../Card";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

describe("Card  component", () => {
  const renderedData = renderer.create(<Card />).toJSON();

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Card />, div);
  });

  it("renders a snapshot", () => {
    expect(renderedData).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
