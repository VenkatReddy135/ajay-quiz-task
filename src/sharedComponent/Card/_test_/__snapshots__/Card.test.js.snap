// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card  component renders a snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": "10em",
    }
  }
>
  <div
    className="card"
    style={
      Object {
        "margin": "auto",
        "width": "20rem",
      }
    }
  >
    <div
      className="card-body"
    >
      <h2
        className="card-title"
      >
        
      </h2>
      <div
        className="card-text"
      />
    </div>
  </div>
</div>
`;
